packages/react-table/src/components/Table/examples/TableFavoritable.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { Table, TableHeader, TableBody, sortable, TableProps } from '@patternfly/react-table';
+import { Checkbox } from '@patternfly/react-core';
+
+interface Repository {
+  name: string;
+  branches: string;
+  prs: string;
+  workspaces: string;
+  lastCommit: string;
+}
+
+export const TableFavoritable: React.FunctionComponent = () => {
+  // In real usage, this data would come from some external source like an API via props.
+  const repositories: Repository[] = [
+    { name: 'one', branches: 'two', prs: 'a', workspaces: 'four', lastCommit: 'five' },
+    { name: 'a', branches: 'two', prs: 'k', workspaces: 'four', lastCommit: 'five' },
+    { name: 'p', branches: 'two', prs: 'b', workspaces: 'four', lastCommit: 'five' }
+  ];
+
+  // Index of the currently sorted column
+  // Note: if you intend to make columns reorderable, you may instead want to use a non-numeric key
+  // as the identifier of the sorted column. See the "Compound expandable" example.
+  const [activeSortIndex, setActiveSortIndex] = React.useState<number | null>(null);
```

Comment:
This may be a matter of opinion but I think `null` is correct here. It represents a "defined empty value" in my mind.

I imagine "no sort applied" as a valid defined sort state, if we had a list of all possible sort states it would be on the list. I would interpret `undefined` to mean "there is no sort state" versus `null` which is "the defined sort state is none". If we had a "clear sort" button, `setActiveSortIndex(undefined)` would feel weirder than `setActiveSortIndex(null)`.

As opposed to the `variant` props above, which should either be passed as a string or not passed at all (`undefined`).

I'm probably being pedantic here though. If people feel strongly that these should be undefined I'm ok with that.

Reply:
It totally makes sense, and thanks for explaining it so well - I was mostly look at it from a tooling standpoint and how the examples transfer as-is when strict mode is enforced. Just as back in the day it was common to enforce all the default rules that ship with the `use strict` directive, it seems like we could benefit from the same approach (enforcing all TypeScript's `strict` mode rules) as it would make the code a little friendlier to copy/paste across projects that use stricter settings than we do here. It is my preference but I'm not dead set on it, I had to make those changes in order to test the examples in my environment, so felt compelled to at least mention it :P